tests/main.cpp

```diff
@@ -712,8 +711,45 @@ TEST_CASE("getVersion") {
 }
 
 TEST_CASE("getFirstAvailableBlock") {
-  const solana::Keypair keyPair = solana::Keypair::fromFile(KEY_PAIR_FILE);
   const auto connection = solana::rpc::Connection(solana::DEVNET);
   const auto firstAvailableBlock = connection.getFirstAvailableBlock();
   CHECK_GT(firstAvailableBlock, 0);
 }
+
+TEST_CASE("getSlot") {
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto slot = connection.getSlot();
+  CHECK_GT(slot, 0);
+  sleep(2);  // sleep for 2 seconds
+  const auto slot2 = connection.getSlot();
+  CHECK_GE(slot2, slot);
+}
+
+TEST_CASE("getSlotLeader") {
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto slotLeader = connection.getSlotLeader();
+  boost::regex expression{"^(?!.*[0OlI_])\\w*$"};
+  CHECK_EQ(boost::regex_match(slotLeader, expression), true);
+}
+
+TEST_CASE("minimumLedgerSlot") {
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto slot = connection.getminimumLedgerSlot();
+  CHECK_GT(slot, 0);
+}
+
+TEST_CASE("getGenesisHash") {
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  const auto hash = connection.getGenesisHash();
+  CHECK_GT(hash.length(), 0);
```

Comment:
Can check with devnet genesis hash. This info is public or you could find it easily using solana cli.

Reply:
Done